test/test_autograd.py

```diff
@@ -435,7 +435,7 @@ def fn(x):
 
         # .backward(inputs=) is OK
         out = c.sum()
-        torch.autograd.backward(out, inputs=(a,), retain_graph=True)
+        torch.autograd.backward(out, inputs=(a, b2), retain_graph=True)
```

Comment:
b2 is not part of the graph, but is passed to backward as an input. When the hook is run, we validate that _will_engine_execute_node returns False.